src/components/sbb-file-selector/sbb-file-selector.tsx

```diff
@@ -0,0 +1,296 @@
+import {
+  Component,
+  ComponentInterface,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  JSX,
+  Prop,
+  State,
+} from '@stencil/core';
+import { InterfaceSbbFileSelectorAttributes } from './sbb-file-selector.custom';
+import {
+  createNamedSlotState,
+  documentLanguage,
+  HandlerRepository,
+  languageChangeHandlerAspect,
+  namedSlotChangeHandlerAspect,
+} from '../../global/eventing';
+import {
+  i18nFileSelectorButtonLabel,
+  i18nFileSelectorDeleteFile,
+  i18nFileSelectorSubtitleLabel,
+} from '../../global/i18n';
+import { toggleDatasetEntry } from '../../global/dom';
+import { sbbInputModalityDetector } from '../../global/a11y';
+
+/**
+ * @slot error - Use this to provide a `sbb-form-error` to show an error message.
+ */
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-file-selector.scss',
+  tag: 'sbb-file-selector',
+})
+export class SbbFileSelector implements ComponentInterface {
+  /** Whether the component has a dropzone area or not. */
+  @Prop() public variant: InterfaceSbbFileSelectorAttributes['variant'] = 'default';
+
+  /** Whether more than one file can be selected. */
+  @Prop() public multiple: boolean;
+
+  /** Whether the newly added files should override the previously added ones. */
+  @Prop() public multipleMode: InterfaceSbbFileSelectorAttributes['multipleMode'];
+
+  /** A comma-separated list of allowed unique file type specifiers. */
+  @Prop() public accept: string;
+
+  /** The title displayed in `dropzone` variant. */
+  @Prop() public titleContent?: string;
+
+  /** Whether the component is disabled. */
+  @Prop() public disabled: boolean;
+
+  /** The list of selected files. */
+  @State() public files: File[];
```

Comment:
```suggestion
  @State() private _files: File[];
```

Reply:
Changed in [58e6011](https://github.com/lyne-design-system/lyne-components/pull/1894/commits/58e6011496b77e847dde243ffed0a60e45915757)